src/System.Text.Json/src/System/Text/Json/Serialization/Converters/JsonValueConverterEnum.cs

```diff
@@ -39,17 +49,54 @@ public override bool TryRead(Type valueType, ref Utf8JsonReader reader, out TVal
                 return false;
             }
 
-            if (s_isUint64)
+            if (s_isUInt64 && reader.TryGetUInt64(out ulong uint64))
             {
-                if (reader.TryGetUInt64(out ulong ulongValue))
-                {
-                    value = (TValue)Enum.ToObject(valueType, ulongValue);
-                    return true;
-                }
+                value = (TValue)Enum.ToObject(valueType, uint64);
+                return true;
+            }
+
+            if (s_isInt64 && reader.TryGetInt64(out long int64))
+            {
+                value = (TValue)Enum.ToObject(valueType, int64);
+                return true;
+            }
+
+            if (s_isUInt32 && reader.TryGetUInt32(out uint uint32))
+            {
+                value = (TValue)Enum.ToObject(valueType, uint32);
+                return true;
+            }
+
+            if (s_isInt32 && reader.TryGetInt32(out int int32))
```

Comment:
Given this is the default and most common, let's do this check first.

Reply:
done